### PR TITLE
[PROD-17650] Expose callback option for which fields will be logged

### PR DESCRIPTION
**Pull request by klesgidis**
## Summary
Expose callback option for which fields will be logged

Usage can be found on documentation

### CODE REVIEW

File: lib/transformers/transformers.js [hunk 1]
```diff
@@ -8,7 +8,15 @@ const url = require('url');
...
 
-const mapValuesDeep = (v, callback) => (isObject(v) ? mapValues(v, v => mapValuesDeep(v, callback)) : callback(v));
+const mapValuesDeep = (v, callback) => {
+  if (v && Array.isArray(v)) {
+    return v.map(vv => mapValuesDeep(vv, callback));
+  } else if (isObject(v)) {
+    return mapValues(v, vv => mapValuesDeep(vv, callback));
+  } else {
+    return callback(v);
+  }
```
Comment (nikostoulas) [inline lib/transformers/transformers.js:18]: ❓ This is noop right? It worked this way before correct?

**Review by nikostoulas — APPROVED** [review]
🚀

**Review by nikosd23 — APPROVED** [review]
👍

File: lib/transformers/transformers.js (hunk 1 above)
Comment (klesgidis) [reply lib/transformers/transformers.js:18]: No it wasn't. Well, It depends 😛 

It was converting arrays to 
```
{
    '0': {...},
    '1': {...},
    '2': {...}
}
```

File: .github/workflows/main.yml [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+# A workflow run is made up of one or more jobs that can run sequentially or in parallel
+jobs:
+  # This workflow contains a single job called "build"
+  build:
+    # The type of runner that the job will run on
+    runs-on: ubuntu-latest
+
+    strategy:
+      matrix:
+        node-version: [8.x, 9.x, 10.x]
```
Comment (nikostoulas) [inline .github/workflows/main.yml:22]: @klesgidis aren't all of these versions deprecated now?

**Pull request merged**